user-operator/src/main/java/io/strimzi/operator/user/operator/batching/AbstractBatchReconciler.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.user.operator.batching;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Abstract class for collecting Kafka Admin API requests and sending them to Kafka in batches. The batches are sent
+ * when we collect some (configurable) amount of requests or after some (configurable) time interval
+ */
+public abstract class AbstractBatchReconciler<T> {
+    private final static Logger LOGGER = LogManager.getLogger(AbstractBatchReconciler.class);
+
+    private final BlockingQueue<T> queue;
+    private final int maxBatchSize;
+    private final int maxBatchTime;
+    private final Thread controllerThread;
```

Comment:
isn't `controller` too much overloaded term? maybe we can use `batchHandlerThread` or something like this?